[PIE-2212] Add docs for team assignment and badges (#2672)
buildkite/docs: pages/test_analytics/flaky_test_assignment.md

# Flaky test assignment

> Customers on the [Business and Enterprise plans](https://buildkite.com/pricing) can assign flaky tests to [Teams](/docs/team_management/permissions).

Reviewer: I see on that pricing page the plan names are 'pro' and 'enterprise'. Should we use the name 'pro' instead of 'business'?

Author: Ah yep, I'd just copied it from the Teams docs, have updated.

# Flaky test assignment

> Customers on the [Pro and Enterprise plans](https://buildkite.com/pricing) can assign flaky tests to [Teams](/docs/team_management/permissions).

Reviewer 2: Whoops, I forgot one more fix...

```suggestion
Customers on the [Pro and Enterprise plans](https://buildkite.com/pricing) can assign flaky tests to [Teams](/docs/team-management/permissions).
```

While our docs system resolves underscores to hyphens, it's probably better to use the real deal here.

## Enabling flaky test assignments

> To enable assignments you must have at least one team that has access to your suite. You may need to ask your admin to enable the teams feature, and then create teams via the organization settings page.

Reviewer 2: ```suggestion
To enable assignments, you must have at least one team that has access to your suite. You may need to ask your admin to enable the teams feature, and then create teams via the organization settings page.
```

<%= image "flaky-test-no-teams.png", width: 1960/2, height: 630/2, alt: "Flaky test page showing dropdown menu prompting user to create teams" %>
<%= image "team-settings.png", width: 1960/2, height: 630/2, alt: "Team page for assigning test suites" %>

## Assigning a test

When viewing your flaky tests, you should now see a list of teams with suite access permissions listed inside the assignment dropdown. From here you may assign, reassign or remove the assignment.

>🚧 Assignment permissions

> > All team members have the ability to create, update or remove an assignment, this feature is not restricted to admins.

Reviewer 2: ```suggestion
> All team members have the ability to create, update or remove an assignment. This feature is not restricted to admins.
```

> Tests that are assigned to a team will update to display a badge indicating as such. This badge is also visible on issues in the run page.

Reviewer 2: ```suggestion
Tests that are assigned to a team will be updated to display a badge indicating as such. This badge is also visible on issues in the run page.
```

Reason: The verb 'update' is transitive and requires an object when used in a sentence/clause.

<%= image "flaky-test-teams.png", width: 1960/2, height: 630/2, alt: "Flaky test page showing team assignments" %>

## Viewing assignments

> Users can check their test assignments by clicking "My Assignments" in the side bar.

Reviewer 2: ```suggestion
Users can check their test assignments by clicking _My Assignments_ in the side bar.
```

Reason: https://github.com/buildkite/docs/blob/main/styleguides/markdown-syntax-style.md#ui-elements

<%= image "current-assignments.png", width: 1960/2, height: 630/2, alt: "Flaky test page showing team assignments" %>

> When an assigned test has not flaked in more than 7 days, it is moved to the "Stale assignment" section. An assignment could become stale due to a flaky test being fixed, or perhaps it belongs to a pipeline which has not had a build in the last 7 days. Should the flake reoccur, the assignment will be moved back to the "Current assignments" page.

Reviewer 2: ```suggestion
When an assigned test has not flaked in more than 7 days, it is moved to the _Stale assignment_ section. An assignment could become stale due to a flaky test being fixed, or perhaps it belongs to a pipeline which has not had a build in the last 7 days. Should the flake reoccur, the assignment will be moved back to the _Current assignments_ page.
```

Same reason above.
